src/js/view/controls/controls.js

```diff
@@ -259,6 +271,22 @@ export default class Controls {
         this.playerContainer.addEventListener('keydown', handleKeydown);
         this.keydownCallback = handleKeydown;
 
+        // If not mobile, show controls after closing overlay
+        if (!OS.mobile) {
+            const blurHandler = (evt) => {
+                const insidePlayer = evt.currentTarget.contains(evt.relatedTarget);
+                const overlayClosed =
+                    evt.currentTarget.classList.contains('jw-flag-overlay-open-related') ||
+                    settingsMenu.element().contains(evt.target);
+
+                // show controls if inside the player and after overlay is closed
+                this.userActive(null, insidePlayer || overlayClosed);
+            };
+
+            this.playerContainer.addEventListener('blur', blurHandler, true);
```

Comment:
So anytime an element in the player looses focus, we check if the `relatedTarget` is inside the player (isn't it always if the listener is on the player)?

Please clarify on blur:
`evt.relatedTarget` (getting focus?)
`evt.currentTarget` (loosing focus?)
`evt.target` (shouldn't this be `playerContainer` or the same as `currentTarget`?)

What does `settingsMenu.element()` have to do with this? It's not covered in your comments.